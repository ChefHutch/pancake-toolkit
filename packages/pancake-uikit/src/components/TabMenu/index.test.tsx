import React from "react";
import userEvent from "@testing-library/user-event";
import { screen } from "@testing-library/react";
import { renderWithTheme } from "../../testHelpers";
import { TabMenu, Tab } from ".";

const handleClick = jest.fn();

it("renders correctly", () => {
  const { asFragment } = renderWithTheme(
    <TabMenu activeIndex={0} onItemClick={handleClick}>
      <Tab>Item 1</Tab>
      <Tab>Item 2</Tab>
    </TabMenu>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should render the correct number of tabs", () => {
  const { getAllByRole } = renderWithTheme(
    <TabMenu activeIndex={0} onItemClick={handleClick}>
      <Tab>Item 1</Tab>
      <Tab>Item 2</Tab>
      <Tab>Item 3</Tab>
    </TabMenu>
  );

  expect(getAllByRole("button").length).toEqual(3);
});

it("should receive the tab index when clicked", () => {
  renderWithTheme(
    <TabMenu activeIndex={0} onItemClick={handleClick}>
      <Tab>Item 1</Tab>
      <Tab>Item 2</Tab>
      <Tab>Item 3</Tab>
    </TabMenu>
  );

  userEvent.click(screen.getByText("Item 2"));
  expect(handleClick.mock.calls[0][0]).toBe(1);
});
