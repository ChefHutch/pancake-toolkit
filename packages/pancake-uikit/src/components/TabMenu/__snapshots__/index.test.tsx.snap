// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border-bottom: 2px solid #8f80ba;
  padding: 0 16px;
}

.c1 > button + button {
  margin-left: 4px;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border: 0;
  outline: 0;
  width: 50%;
  min-width: 83px;
  padding: 8px 0;
  border-radius: 16px 16px 0 0;
  color: #FFFFFF;
  background-color: #8f80ba;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border: 0;
  outline: 0;
  width: 50%;
  min-width: 83px;
  padding: 8px 0;
  border-radius: 16px 16px 0 0;
  color: #8f80ba;
  background-color: #eeeaf4;
}

.c3 {
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0 8px;
  font-weight: 600;
}

.c5 {
  color: #8f80ba;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0 8px;
  font-weight: 600;
}

@media screen and (min-width:576px) {
  .c2 {
    width: auto;
  }
}

@media screen and (min-width:576px) {
  .c4 {
    width: auto;
  }
}

<div
    class="c0 c1"
  >
    <button
      class="c2"
      color="card"
    >
      <div
        class="c3"
        color="card"
        font-weight="600"
      >
        Item 1
      </div>
    </button>
    <button
      class="c4"
      color="textSubtle"
    >
      <div
        class="c5"
        color="textSubtle"
        font-weight="600"
      >
        Item 2
      </div>
    </button>
  </div>
</DocumentFragment>
`;
